routing/localchans/manager.go

```diff
@@ -106,6 +127,38 @@ func (r *Manager) UpdatePolicy(newSchema routing.ChannelPolicy,
 		return err
 	}
 
+	// Return an error if any of the channels in chansToUpdate
+	// were invalid.
+	var errStr string
+	for chanPoint := range chansToUpdate {
+		channel, err := r.FetchChannel(chanPoint)
+		if errors.Is(err, channeldb.ErrChannelNotFound) {
+			errStr += fmt.Sprintf(
+				"channel %s:%d not found; ",
+				chanPoint.Hash.String(), chanPoint.Index)
+			continue
+		}
+		if err != nil {
+			errStr += fmt.Sprintf(
+				"error fetching channel %s:%d; ",
+				chanPoint.Hash.String(), chanPoint.Index)
+			continue
+		}
+		if channel.IsPending {
+			errStr += fmt.Sprintf("channel %s:%d is "+
+				"not yet confirmed; ",
+				chanPoint.Hash.String(), chanPoint.Index)
+			continue
+		}
+		errStr += fmt.Sprintf("channel %s:%d could "+
+			"not update its policies; ",
+			chanPoint.Hash.String(), chanPoint.Index)
+	}
+	if errStr != "" {
+		return fmt.Errorf("policy update failed: %s %s",
+			errStr, errInactiveChans)
+	}
+
```

Comment:
there is still the issue here that if some channels were updated successfully and some were invalid then it will return here and not call `r.UpdateForwardingPolicies(policiesToUpdate)` below. I think you can get rid of `errStr` and just rather append these all to `errInvalidChans`.